types/convert.go

```diff
@@ -217,7 +217,7 @@ func NumberToDuration(number int64, fsp int) (Duration, error) {
 	}
 
 	if number/10000 > TimeMaxHour || number%100 >= 60 || (number/100)%100 >= 60 {
-		return ZeroDuration, ErrInvalidTimeFormat
+		return ZeroDuration, errors.Trace(ErrInvalidTimeFormat.GenByArgs(number))
```

Comment:
no need to `Trace`, it's already the root cause.

Reply:
I'm afraid this is necessary, without it we'll lost the last stack

```
package main

import (
	"fmt"

	"github.com/juju/errors"
)

func main() {
	e := h()
	fmt.Println(errors.ErrorStack(e))
}

type MyE struct{}

func (_ MyE) Error() string {
	return "hello"
}

func h() error {
	return errors.Trace(g())
}

func g() error {
	return errors.Trace(f())
}

func f() error {
	if false {
		return MyE{}  
	}
	return MyE{} 
}
```

The result is:

```
➜  test go run test.go
hello
/home/genius/project/test/test.go:25: 
/home/genius/project/test/test.go:21: 
```

f() may return two errors, we don't know which one without line information.
Change f() to 

```
func f() error {
	if false {
		return errors.Trace(MyE{})
	}
	return errors.Trace(MyE{})
}
```

```
➜  test go run test.go
hello
/home/genius/project/test/test.go:32: 
/home/genius/project/test/test.go:25: 
/home/genius/project/test/test.go:21:
```

We know exactly it's line 32, not line 30...

Reply:
ok